### PR TITLE
Update freebayes to 1.2.0

### PR DESCRIPTION
**Pull request by npavlovikj**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/freebayes/meta.yaml [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
   git_url: https://github.com/ekg/freebayes.git
-  git_tag: 8d2b3a060da473e1f4f89be04edfce5cba63f1d3
+  git_tag: 40155b407a4bead708aaafcadf7904854c411275
   patches:
     - qcfail.diff
 
 build:
-  number: 3
-  skip: true  # [osx]
+  number: 4
```
Comment (Slugger70) [inline recipes/freebayes/meta.yaml]: Reset the build number to `0` for a new version of the software.

Comment (npavlovikj) [reply recipes/freebayes/meta.yaml]: Thanks @Slugger70  - done!

File: recipes/freebayes/meta.yaml [hunk 2]
```diff
@@ -1,16 +1,15 @@
 package:
   name: freebayes
-  version: '1.1.0.46'
+  version: '1.2.0'
 
 source:
   git_url: https://github.com/ekg/freebayes.git
-  git_tag: 8d2b3a060da473e1f4f89be04edfce5cba63f1d3
+  git_tag: 40155b407a4bead708aaafcadf7904854c411275
```
Comment (bgruening) [inline recipes/freebayes/meta.yaml]: It would be good to use the official release tarball next time.

Comment (npavlovikj) [reply recipes/freebayes/meta.yaml]: I was planning of using the tarball @bgruening , but "freebayes" has few submodule dependencies that are not part of the official release. I plan to make recipes for the missing dependencies, but that may take some time.

**Pull request merged**